src/api/main/NativeInterfaceFactory.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+import type {NativeInterfaceMain} from "../../native/main/NativeInterfaceMain"
+import type {NativePushServiceApp} from "../../native/main/NativePushServiceApp"
+import type {NativeSystemApp} from "../../native/main/NativeSystemApp"
+import type {NativeFileApp} from "../../native/common/FileApp"
+import {downcast} from "@tutao/tutanota-utils"
+import {isBrowser} from "../common/Env"
+import {ProgrammingError} from "../common/error/ProgrammingError"
+
+/**
+ * A factory function to create interfaces for native communication
+ * When running in browser, these interfaces will be stubs that throw if you try to use them
+ */
+export async function createNativeInterfaces(): Promise<{
+	native: NativeInterfaceMain,
+	fileApp: NativeFileApp,
+	pushService: NativePushServiceApp,
+	systemApp: NativeSystemApp
+}> {
+
+	let native
+	let fileApp
+	let pushService
+	let systemApp
+
+	if (!isBrowser()) {
+		const {NativeInterfaceMain} = await import("../../native/main/NativeInterfaceMain")
+		const {NativeFileApp} = await import ("../../native/common/FileApp")
+		const {NativePushServiceApp} = await import("../../native/main/NativePushServiceApp")
+		const {NativeSystemApp} = await import("../../native/main/NativeSystemApp")
+
+		native = new NativeInterfaceMain()
+		fileApp = new NativeFileApp(native)
+		pushService = new NativePushServiceApp(native)
+		systemApp = new NativeSystemApp(native, fileApp)
+	} else {
+		/** Trying to access native interfaces from the browser is an error */
```

Comment:
Ideally we shouldn't even inject any of them when we don't need them but without rewriting a lot of code this might work temporarily. If we keep it, we should remove TODO.
Alternative is to not call it on web and instead throw in the getter of locator (either with `defineProperty` or separating public and private fields).